Upgrades for Qualimap BamQC module

vladsaveliev opened the pull request:
Two proposed improvements in 2 commits:

1). Add a line for human reference to the GC content distribution plot.
<img width="1062" alt="screen shot 2016-10-30 at 01 18 07" src="https://cloud.githubusercontent.com/assets/1575412/19833069/24a0e7a2-9e3e-11e6-934c-7ba5a88a0bce.png">

2). Draw cumulative coverage ("Genome Fraction Coverage") plot for values above 50x, and align it vertically with the coverage histogram.

Currently both plots' X axes are cut independently. The histogram is cut at values below 1% to prevent long flat tail. "Genome Fraction" is cut at 50x because Qualimap doesn't provide more data for it:

<img width="1118" alt="screen shot 2016-10-30 at 01 22 13" src="https://cloud.githubusercontent.com/assets/1575412/19833088/aefb5798-9e3e-11e6-8b47-9746d4b5674e.png">

I propose to calculate cumulative coverage from the histogram data, so it will contain values above 50x; and cut it at exact same threshold:

<img width="1102" alt="screen shot 2016-10-30 at 01 19 31" src="https://cloud.githubusercontent.com/assets/1575412/19833075/4dd49c90-9e3e-11e6-9a8c-d16c16389a24.png">

ewels commented:
Hi @vladsaveliev,

This is great, thanks! The absence of data above 50X has always annoyed me a little, I'd not realised that we have the data available to recalculate this ourselves - super nice! Also it's always good to reduce the number of files we need to parse.

A few quick questions from my first impressions:

1. Human reference to the GC content distribution plot
  * Is this data _always_ present in the file? I have a vague memory of it being an optional parameter to calculate this in Qualimap meaning that the column isn't always present. I may be wrong though.
  * What happens when the data is non-Human? Does Qualimap still generate this curve for the reference?
2. No comments - looks great!

I'm hoping to have some time to work with MultiQC next week all being well. I'll have a proper go through your code then.

Phil

vladsaveliev commented:
Thanks Phil!

You are right, the reference GC data is actually only reported if Qualimap was run with the `-gd` parameter, and the supported parameters are only HUMAN and MOUSE. I'm adding a fix so it will only reported if the reference GC column is present.

ewels commented:
Great! Thanks for fixing the Py3 errors too.

So... What happens if people run MultiQC on a bunch of Qualimap reports that are both Mouse and Human? 😉  (the more popular this tool becomes, the more paranoid about edge cases I become!)

vladsaveliev commented:
Good point, I'll address this special case as well. It's going to be 2 extra plots. Luckily, mapped_reads_gc-content_distribution.txt reports a header specifying if this is the data for human or mouse.

If you have more weird edge cases in mind, please don't hesitate to share :)

Would you mind if I squash commits once I'm finished? Keeping one for GC and one for coverage.

ewels commented:
@vladsaveliev - I've read through your code and given it a quick test and it all looks great to me. I haven't been able to test the GC content line as https://github.com/ewels/MultiQC_TestData doesn't have any files with that data, but I'm happy to trust you that it works (if you fancy submitting a PR there with such example data, that would be even better).

I'll leave you to do any commit squashing that you want to do, just let me know when you're ready for me to merge. Also, it would be great if you could add to the changelog and add yourself to the list of contributors at the bottom of the readme.. :)

Thanks again, I think we'll find your changes very helpful!

ewels commented:
Oh, one minor thing - it would be nice to add a short bit of text above the GC plot when a dotted line is present explaining what it shows.

vladsaveliev commented:
Okay, I'm finished with my changes. I added a bit explaining the GC dotted line, added test data to MultiQC_TestData, also fixed a bug where QualiMap strips the sample name out of the base folder (`sname.rstrip('.qc')` removes any letter from the list, so `"with_human_gc".rstrip('.qc') == "with_human_g"`). I also put all BamQC titles to a common standard with regards to the uppercase/lowercase; that bothered my perfectionism a bit, so let me know if you are not okay with that.

I rebased my commits and put all of them above yours, so the PR is ready to be merged now without any conflicts. 

Thanks for accepting my improvements and especially for your suggestions and review!

ewels commented:
Sounds great, thanks for your work!
